dasharo-security/measured-boot.robot

```diff
@@ -18,25 +18,376 @@ Resource            ../keys.robot
 # - document which setup/teardown keywords to use and what are they doing
 # - go threough them and make sure they are doing what the name suggest (not
 # exactly the case right now)
-Suite Setup         Run Keyword
-...                     Prepare Test Suite
+Suite Setup         Measured Boot Suite Setup
 Suite Teardown      Run Keyword
 ...                     Log Out And Close Connection
 
 
+*** Variables ***
+${PCRS_TO_CHECK}=       [0-9]|14
+
+
 *** Test Cases ***
 MBO001.001 Measured Boot support (Ubuntu 20.04)
     [Documentation]    Check whether Measured Boot is functional and
-    ...    measurements are stored into the TPM.a.
-    Skip If    not ${MEASURED_BOOT_SUPPORT}    MBO001.001 not supported
-    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    MBO001.001 not supported
+    ...    measurements are stored into the TPM.
+    Execute Command In Terminal    shopt -s extglob
+    ${hashes}=    Execute Command In Terminal
+    ...    grep . /sys/class/tpm/tpm0/pcr-sha@(1|256)/[0-3]
+    Should Not Contain Any    ${hashes}
+    ...    0000000000000000000000000000000000000000
+    ...    FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
```

Comment:
Not an imminent danger of a false positive, but I'd check for sha1 and sha256 separately to not assume that no sha256 hash has a sequence of 40 0s or Fs inside of its 64-character value.

Reply:
fixed in [802f7e5](https://github.com/Dasharo/open-source-firmware-validation/pull/326/commits/802f7e57553d1ff09c92255ae050c4d06e75f8b6)